src/components/cylc/Graph.vue

```diff
@@ -0,0 +1,1284 @@
+  animate: false, // whether to animate on drawing changes you can specify a function too
+  ready: function () {}, // callback when expand/collapse initialized
+  undoable: true, // and if undoRedoExtension exists,
+  cueEnabled: true, // Whether cues are enabled
+  expandCollapseCuePosition: 'top-left', // default cue position is top left you can specify a function per node too
+  expandCollapseCueSize: 20, // size of expand-collapse cue
+  expandCollapseCueLineSize: 16, // size of lines used for drawing plus-minus icons
+  expandCueImage: undefined, // image of expand icon if undefined draw regular expand cue
+  collapseCueImage: undefined, // image of collapse icon if undefined draw regular collapse cue
+  expandCollapseCueSensitivity: 1 // sensitivity of expand-collapse cues
+}
+
+const expandCollapseOptionsCola = {
+  layoutBy: {
+    name: 'cose-bilkent',
+    animate: 'end',
+    randomize: false,
+    fit: false
+  },
+  // recommended usage: use cose-bilkent layout with randomize: false to preserve mental map upon expand/collapse
+  fisheye: false, // whether to perform fisheye view after expand/collapse you can specify a function too
+  animate: false, // whether to animate on drawing changes you can specify a function too
+  ready: function () {}, // callback when expand/collapse initialized
+  undoable: true, // and if undoRedoExtension exists,
+  cueEnabled: true, // Whether cues are enabled
+  expandCollapseCuePosition: 'top-left', // default cue position is top left you can specify a function per node too
+  expandCollapseCueSize: 20, // size of expand-collapse cue
+  expandCollapseCueLineSize: 16, // size of lines used for drawing plus-minus icons
+  expandCueImage: undefined, // image of expand icon if undefined draw regular expand cue
+  collapseCueImage: undefined, // image of collapse icon if undefined draw regular collapse cue
+  expandCollapseCueSensitivity: 1 // sensitivity of expand-collapse cues
+}
+
+export default {
+  name: 'Graph',
+  props: ['workflowid'],
+  data: function () {
+    return {
+      isConnected: false,
+      config,
+      elements,
+      graphData: {
+        nodes: [],
+        edges: []
+      },
+      i: 1,
+      // vue-spinner
+      color: '#5e9aff',
+      height: '35px',
+      width: '4px',
+      margin: '2px',
+      radius: '2px',
+      size: '1em',
+      loading: true,
+      // layout variables
+      status: 'pending',
+      layoutName: 'dagre',
+      layoutStopped: true,
+      layoutReady: false,
+      // elements
+      nodesEdges: [],
+      workflows: [],
+      edges: [],
+      subscriptions: {},
+      workflowId: ''
+    }
+  },
+  watch: {
+    graphData: debounce(function (newVal) {
+      this.debouncer.call()
+    }, 100),
+
+    layoutName: {
+      handler (value) {
+        this.updateLayout(value)
+      }
+    },
+
+    workflows: {
+      handler: function (newval, oldval) {
+        this.workflowUpdated(newval)
```

Comment:
With watch I can use `deep` to be aware of data changes, this is useful because the data is very nested. I was hoping it wouldn't be complex nested data, then I could use computed on just the graph data.